### DISCUSSION TITLE
Stop http redirect

### DISCUSSION THREAD

**Discussion by f3rf4** [pashpashpash/vault-ai#36] [Q&A] [answered]
Hi,
How do I stop the web server from redirecting to https. When I go to the ip address and port number of the site then it redirects to https and the site won't load. Any help would be appreciated. Thanks.

**Comment by pashpashpash** [comment]
@f3rf4 This should only happen when you start the server with flag -port 443? (`npm start -- -port 443`) This shouldn't happen if you're starting without that flag: `npm start` (for localhost) or `npm start -- -port 80` (for http)

**Reply by f3rf4** [reply]
I tried both npm start (for localhost) and npm start -- -port 80 and it's still redirecting to https and wont load the site.

**Reply by pashpashpash** [reply]
@f3rf4 
Try removing this line:
https://github.com/pashpashpash/vault-ai/blob/master/vault-web-server/main.go#L92

**Reply by f3rf4** [reply]
That doesn't work either. Still redirecting. I appreciate the help on this.

**Reply by Silver556** [reply]
Any luck figuring out how to prevent the HTTPS redirect?

If its not possible/simple to prevent the HTTPS redirect, do lines 93 & 94 need to be modified to be appropriate for the deployment?

certFile := "/etc/letsencrypt/live/**vault.pash.city**/fullchain.pem"
keyFile := "/etc/letsencrypt/live/**vault.pash.city**/privkey.pem"

https://github.com/pashpashpash/vault-ai/blob/master/vault-web-server/main.go#L93

**Reply by trv893** [reply]
would also love an answer to this!
im trying to access the site via the ip address from my local computer with it running on an ubuntu vm and keep getting redirected.
cheers!

**Reply by bannedin40countries** [reply, marked as answer]
Remove the ghetto https redirect. That works for me:

https://github.com/pashpashpash/vault-ai/blob/bcf02b421513fd43e1ce9ff41d342267e41e10d5/static/index.html#L41C5-L53

**Reply by f3rf4** [reply]
Just that line the whole script?

**Reply by bannedin40countries** [reply]
> Just that line the whole script?

Yes, remove the whole script.

The script checks whether the hostname of the current location is not "localhost" and the protocol is not "https". If this condition is true, it replaces the current location with the same URL but with "https" protocol. In other words, it redirects the user to https. Hope that works for you.

**Reply by f3rf4** [reply]
That works great.  I tried messing with all kinds of stuff in the main.go file but couldn't get it to work. Thankyou very much.
I also get an error 500 runtime error when uploading a file to it. Do you know why that is?
